packages/core/src/chakra-provider.tsx

```diff
@@ -45,6 +46,14 @@ export const ChakraProvider: React.FC<ChakraProviderProps> = (props) => {
     throw Error("ChakraProvider: the `theme` prop is required")
   }
 
+  Object.keys(theme.components).forEach((component) => {
+    return theme.components[component].parts
```

Comment:
besides `return` doing nothing in a forEach, I'd prefer this out of readability and testability reasons:

```ts
const applyStyleConfigs = (theme: Theme) => {
  Object.keys(theme.components).forEach((component) => {
    const config = theme.components[component];

    if(config.parts) {
      theme.components[component] = multiStyleConfig(config);
    } else {
      theme.components[component] = styleConfig(config)
    } 
  });

 return theme;
}
```

Reply:
we might keep `applyStyleConfigs` in `theme-tools` then too, importing `multiStyleConfig`/`styleConfig` would be internal then I think

Reply:
@ljosberinn what do you think of writing this function in `@chakra-ui/theme-tools` (`theme-tools/src/style-config`)?

Reply:
sounds perfect

Reply:
@ljosberinn I made some changes and created the function. Does it seem good to you?

Reply:
Also, what do you think about changing the theme type in the `ThemeProvider`?

From `Dict`
```Typescript
export interface ThemeProviderProps {
  /**
   * The theme to use for your application
   */
  theme: Dict
}
```
To `Theme`

```Typescript
import { Theme } from '@chakra-ui/theme'

export interface ThemeProviderProps {
  ...
  theme: Theme
}
```

Reply:
I'm weary of a change of the theme provider, would it still work if you omit parts of the theme, e.g. because you're not using all components and want to reduce bundle size?

Reply:
hmmm got your point, so let's keep it `Dict` for now